### PR TITLE
tests_docs: do not look at CI var

### PR DESCRIPTION
**Pull request by jgehrcke**
This is acting on my thinking in #496 -- would appreciate your review @austin3dickey -- thanks!

### CODE REVIEW

File: conbench/tests/api/test_docs.py [hunk 1]
```diff
@@ -1,38 +1,43 @@
...
+            # Hoping that this shows a useful diff.
+            log.info("caught assertion error: %s", exc)
+            raise Exception(
+                "/api/docs.json and _expected_docs.py are out of sync. "
+                "Run `make rebuild-expected-api-docs` to regenerate "
+                "_expected_docs.py and then review the diff manually."
+            )
+
+        # Maybe this is where we need a Python object structure, and not just
+        # a JSON document.
```
Comment (austin3dickey) [inline conbench/tests/api/test_docs.py:42]: I believe `validate_spec()` is intended as a cheap way to ensure the docs will render correctly using swagger. I'm not sure if ser/de to JSON will break anything here?

File: conbench/tests/api/test_docs.py [hunk 2]
```diff
@@ -1,38 +1,43 @@
...
             expected_docs = ast.literal_eval(f.read())
 
         try:
+            # Note(JP): it seems that the hope here is that `assert_200_ok`
+            # performs a deep object equality inspection, relying on the
+            # `assert r.json == expected` equality check in the implementaion
+            # of `assert_200_ok()`. But how deep is that, really? Typically,
+            # the safest way to make sure that all details are after all equal
+            # is to perform stable string serialization. What would we lose if
+            # we were to compare JSON documents here?
```
Comment (austin3dickey) [inline conbench/tests/api/test_docs.py:30]: All I know is that `assert_200_ok()` is used in dozens of our tests, and never has had a false positive or negative that I'm aware of. So I'm fine with it. But you're welcome to try to replace its internals with a JSON serializer!

**Review by austin3dickey — APPROVED** [review]
looks great!

File: conbench/tests/api/test_docs.py (hunk 1 above)
Comment (jgehrcke) [reply conbench/tests/api/test_docs.py:42]: It makes complete sense that this function wants to have a Python object structure as input. But that could simply be `validate_spec(resp.json)`.

File: conbench/tests/api/test_docs.py (hunk 2 above)
Comment (jgehrcke) [reply conbench/tests/api/test_docs.py:30]: :)
The false negatives are hard to see of course.

I mean, Python has rather deep deep deep comparison when basic data types are used (lists, tuples, dicts), and therefore it is likely that the outest equality operator after all means "compare _all_ the details". But there is no language guarantee.

**Comment by jgehrcke** [comment]
Thanks for the review, @austin3dickey.

For the record, I am wondering what we'd lose if were to do:

```python
resp = client.get("/api/docs.json")
assert resp.data== open("expected_docs.json"), "rb").read()
validate_spec(resp.json)
```

That confirms that `docs.json` represents a valid OpenAPI spec, and that directly confirms that the JSON byte sequence in the response to matches the expected byte sequence byte-by-byte.

The JSON doc should for this be in stable sort order and ideally be pretty-printed to make diffs readable. But I think that's already the case.

I am not really sure what the complexity in `ast.literal_eval(f.read())` and `black` is good for in this context, but I trust that at some point there was value or maybe there _is_, and I don't see it.

Moving on from here, for now :)

**Pull request merged**